code/game/gamemodes/wizard/soulstone.dm

```diff
@@ -106,69 +107,261 @@
+		//TODO: add a few snowflake checks to specific simple_animals that could be soulstoned.
+
+/obj/item/proc/capture_soul_head(var/target, var/mob/user as mob)//called either when using a soulstone on a head, or on a decapitated body
+	if(istype(target, /obj/item/weapon/organ/head))
+		var/obj/item/weapon/organ/head/H = target
+		if(!H.organ_data)
+			to_chat(user, "<span class='rose'>The soul stone isn't reacting, looks like their brain was separated from their head.</span>")
+			return
+		var/mob/living/carbon/brain/T = H.brainmob
+		if(!T.client)
+			if(!T.mind)
+				to_chat(user, "<span class='warning'>The soul stone isn't reacting, looks like this target doesn't have much of a soul.</span>")
+				return
+			else
+				var/mob/new_target = null
+				for(var/mob/M in player_list)
+					if(M.key == T.mind.key)
+						new_target = M
+				if(!new_target)
+					to_chat(user, "<span class='warning'>The soul stone isn't reacting, looks like this target's soul went far, far away.</span>")
+					return
+				else if(!istype(new_target,/mob/dead/observer))
+					to_chat(user, "<span class='warning'>The soul stone isn't reacting, looks like this target's soul already reincarnated.</span>")
+					return
+				else
+					to_chat(new_target, "<span class='danger'>You feel your soul getting sucked into the soulstone.</span>")
+					to_chat(user, "<span class='rose'>The soul stone reacts to the corpse and starts glowing.</span>")
+					capture_soul_process(user,new_target.client,H,H.origin_body)
+		else
+			to_chat(T, "<span class='danger'>You feel your soul getting sucked into the soul stone.</span>")
+			to_chat(user, "<span class='rose'>The soul stone reacts to the corpse and starts glowing.</span>")
+			capture_soul_process(user,T.client,H,H.origin_body)
+
+
+/obj/item/proc/capture_soul_process(var/mob/living/carbon/user, var/client/C, var/atom/movable/target, var/atom/movable/add_target = null)
+	//user is the guy using the soulstone
+	//C is the client of the guy we're soulstoning, so we don't lose track of him between the beginning and the end of the soulstoning.
+	//target is the source of the guy's soul (his body, or his head if decapitated)
+	//add_target is his body if he has been decapitated, for cosmetic purposes (and so it dusts)
+
+	if(!C)
+		return
+
+	var/mob/living/carbon/human/body = null
+
+	if(istype(target,/mob/living/carbon/human))
+		body = target
+	else if(istype(add_target,/mob/living/carbon/human))
+		body = add_target
+
+	var/true_name = "Unknown"
+
+	if(body)
+		true_name = body.real_name
+
+		for(var/obj/item/W in body)
+			body.drop_from_inventory(W)
+
+		body.dropBorers(1)
+
+		var/turf/T = get_turf(body)
+
+		body.invisibility = 101
+
+		var/datum/organ/external/head_organ = body.get_organ("head")
+		if(head_organ.status & ORGAN_DESTROYED)
+			new /obj/effect/decal/remains/human/noskull(T)
+			var/atom/movable/overlay/animation = new /atom/movable/overlay(T)
```

Comment:
Yeah MAYBE that happens with just setting icon_state to a nonlooping animation

But flick() wouldn't have that problem